ui/src/components/ConfigurationForm/ConfigurationForm.helpers.ts

```diff
@@ -17,10 +17,13 @@ export const mapEffectTypeToEffectImplementation = (effectType: EffectType): Eff
   return effectTypeImplementationMap[effectType];
 };
 
-export const getEffectImplementationOptions = (effectType: EffectType): FormSelectOption[] => {
+export const getEffectImplementationOptions = (effectType: EffectType, scalaVer: ScalaVersion): FormSelectOption[] => {
```

Comment:
Maybe a bit more verbose? Ideally this should be piped, I need to add some library to help with this.

```typescript
export const getEffectImplementationOptions = (effectType: EffectType, scalaVer: ScalaVersion): FormSelectOption[] => {
  const mappedEffectImplementations = mapEffectTypeToEffectImplementation(effectType);
  const forbiddenScala3EffectImplementations = [EffectImplementation.Akka];

  const availableEffectImplementations = mappedEffectImplementations.filter(effectImplementation =>
    scalaVer === ScalaVersion.Scala2
      ? effectImplementation
      : !forbiddenScala3EffectImplementations.includes(effectImplementation)
  );

  return EFFECT_IMPLEMENTATIONS_OPTIONS.filter(({ value }) => availableEffectImplementations.includes(value));
};
```

Reply:
Done